File: Subclass/sideFooter.m
//
//  sideFooter.m
//  BeFit
//
//  Created by Jon Brown on 11/30/13.
//
//

#import "sideFooter.h"
#import <QuartzCore/QuartzCore.h>

@implementation sideFooter

- (void)drawRect:(NSRect)dirtyRect {
    
    
//    NSColor *backColor = [NSColor colorWithCalibratedRed:0.80 green:0.83 blue:0.86 alpha:1.0f];
//    
//    [backColor setFill];
//    NSRectFill(dirtyRect);
//    [super drawRect:dirtyRect];
    
    
    CALayer *backgroundLayer = [CALayer layer];
    [backgroundLayer setBackgroundColor:CGColorCreateGenericRGB(0.80, 0.83, 0.86, 1.0f)]; //RGB plus Alpha Channel
    
    
    [self setLayer:backgroundLayer];
    [self setWantsLayer:YES];
    }

@end

File: Subclass/SideFooter.swift
//
//  SideFooter.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/6/19.
//

import Foundation
import Cocoa

class SideFooter: NSView {
    override func draw(_ dirtyRect: NSRect) {
        //    NSColor *backColor = [NSColor colorWithCalibratedRed:0.80 green:0.83 blue:0.86 alpha:1.0f];
        //
        //    [backColor setFill];
        //    NSRectFill(dirtyRect);
        //    [super drawRect:dirtyRect];
            
            
        let backgroundLayer = CALayer()
        backgroundLayer.backgroundColor = NSColor(red: 0.8, green: 0.83, blue: 0.86, alpha: 1).cgColor
            
        layer = backgroundLayer
        wantsLayer = true
    }
}
